### DISCUSSION TITLE
DSS Data Engineer

### DISCUSSION THREAD

**Discussion by dimasrepo** [teamalgoritma/community#915] [Q&A] [answered]
Dags Import Error
![image](https://github.com/user-attachments/assets/0602e0ed-38f3-43c3-b773-7b1937538b56)
![image](https://github.com/user-attachments/assets/76a55c0a-905a-4bdd-8fcd-ef67c4c63a11)
Sebelumnya tidak ada eror saat dicoba kembali tiba tiba eror seperti ini
berikut file dags yang sudah di rubah sesuai guideline
Fungsi fetch_clean()
![image](https://github.com/user-attachments/assets/6ffd42a4-0899-473c-9046-8adb27fedf64)
Fungsi db_to_db() & Fungsi report_generator()
![image](https://github.com/user-attachments/assets/cffb5a9c-d7f5-4b46-a465-5d30b486e3d3)
Untuk pathnya juga sudah sesuai \\wsl.localhost\Ubuntu\home\dimas\airflow\dags
![image](https://github.com/user-attachments/assets/fb6a7254-815f-4b58-bebd-a2acdbb89093)
berikut file dagsnya
[dags.zip](https://github.com/user-attachments/files/16835943/dags.zip)

**Comment by IrfanChairurrachman** [comment]
Dear @dimasrepo ,

Terima kasih telah mengirimkan kodenya. Hanya saja kami masih belum dapat tracing letak errornya di mana. Mungkin, tolong dikirimkan screenshot dari isi folder dags di dalam airflownya Pak? Apakah tampilannya seperti berikut? atau hanya terdapat file dss_dag.py seperti zip yang telah Anda kirimkan?

![image](https://github.com/user-attachments/assets/d672bbdf-4251-4e02-90f4-678521accd05)

**Comment by dimasrepo** [comment]
Berikut mas 
1. Folder dags zip
![image](https://github.com/user-attachments/assets/d25a6a6b-5666-4465-9842-6fc9376210c1)
2. Folder Airflow
![image](https://github.com/user-attachments/assets/5c566d23-37b9-48e6-8253-18c2b6cbca77)
3. Folder dags Airflow
![image](https://github.com/user-attachments/assets/880bd4e9-f65d-4761-8d6b-d306c0282159)

Yang saya pinda hanya file  dss_dag.py saja sih mas ke airflownya. Apa mungkin perlu dipindah semuanya dalam bentuk folder dags ? atau hanya file dalam dagsnya saja ya mas ?

Terimakasih

**Reply by IrfanChairurrachman** [reply, marked as answer]
dear @dimasrepo ,

Yang dipindahkan keseluruhan file dan folder di dalam /dags ya Pak, jadinya yang harus dipindahkan adalah:

- dss_dag.py
- folder data_input_airflow
- folder db
- folder report

Sehingga strukturnya mirip dengan isi folder dari dags.zip.

Kenapa harus dipindahkan, karena dss_dag.py ini butuh untuk membaca data dari `data_input_airflow`, butuh untuk kirim ke database di dalam folder `db` dan butuh untuk membuat report yang disimpan di dalam folder `report`. Maka dari itu harus dipindahkan juga file-file yang dibutuhkan.

Sekian, semoga membantu Pak

**Comment by adizaen** [comment]
Dear @dimasrepo 

Terima kasih telah bertanya melalui Github discussions Algoritma. Untuk menjaga agar forum diskusi berjalan kondusif, karena tidak ada respon lanjutan setelah 7 hari dari respon terakhir, kami izin menandai pertanyaan sudah terjawab dan menutup diskusi ini. Semoga dapat dipahami dan terima kasih.

**Best regards,**
Team Algoritma
